### PR TITLE
Fix module registration.

### LINKED ISSUE #75: Module registration currently doesn't work
In trying to compile an R package using extendr on Windows, I have found that the module registration doesn't work. I believe the problem is in this line:
https://github.com/extendr/extendr/blob/41772722a01062b064320c98d687e3eae889790a/extendr-macros/src/lib.rs#L347

The function to be called should be named `R_init_<modname>`, not `R_init_lib<modname>`. See e.g. here:
https://github.com/r-rust/hellorust/blob/0b53c737ead980f58fe9584fbf513f15fdfe0582/src/wrapper.c#L30

When doing manual, C-style registration the checks work:
https://github.com/extendr/helloextendr/runs/1561836399?check_suite_focus=true

When doing the current module registration the Windows check doesn't work, the package compiles but then the exported function isn't found:
https://github.com/extendr/helloextendr/runs/1561757762?check_suite_focus=true#step:11:251

For some reason, Linux and macOS don't care whether the functions are properly registered or not.

### PR DESCRIPTION
**Pull request by clauswilke**
Fix module registration by forwarding the registration call from C to Rust. Apparently this is needed on Windows i686. See discussion here: #91 

This closes #75.

### CODE REVIEW

**Comment by andy-thomason** [comment]
Worth noting that the `extendr_module!` macro works pretty well now.

**Comment by andy-thomason** [comment]
Ah. I'm a bit behind the times. There is a naming issue with the DLL on windows that you are fixing.

**Comment by clauswilke** [comment]
Yes, correct. On Windows the linker strips all symbols from the generated library unless there's a function called `R_init_<pkgname>()` in the C file. This PR fixes this problem.

File: tests/extendrtests/src/rust/Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ crate-type = ["staticlib"]
 extendr-api = "*"
 
 [patch.crates-io]
-extendr-api = { git = "https://github.com/extendr/extendr" }
-extendr-engine = { git = "https://github.com/extendr/extendr" }
-extendr-macros = { git = "https://github.com/extendr/extendr" }
+extendr-api = { git = "https://github.com/clauswilke/extendr", branch = "reg-forwarding" }
+extendr-engine = { git = "https://github.com/clauswilke/extendr", branch = "reg-forwarding" }
+extendr-macros = { git = "https://github.com/clauswilke/extendr", branch = "reg-forwarding" }
```
Comment (hobofan) [inline tests/extendrtests/src/rust/Cargo.toml:21]: @clauswilke Should this have been merged? Looks like a temporary workaround.

Comment (clauswilke) [reply tests/extendrtests/src/rust/Cargo.toml:21]: It's necessary, see #89. I'll fix with #91 as soon as GH Actions complete.

Comment (hobofan) [reply tests/extendrtests/src/rust/Cargo.toml:21]: 👍 Ah okay, hadn't seen that issue yet.

**Pull request merged**